File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let profit = 0;
        let minPrice = prices[0];
        for (let i = 0; i < prices.length; i++) {
            minPrice = Math.min(minPrice, prices[i]);
            profit = Math.max(profit, prices[i] - minPrice)
        }
        return profit
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans = nums[0];
        int fn = nums[0];
        for (int i = 1; i < nums.length; i++) {
            fn = Math.max(fn + nums[i], nums[i]);
            ans = Math.max(ans, fn);
        }
        return ans;
    }
}
